src/components/application_manager/include/application_manager/application_impl.h

```diff
@@ -610,6 +618,7 @@ class ApplicationImpl : public virtual Application,
   AppFilesMap app_files_;
   std::set<mobile_apis::ButtonName::eType> subscribed_buttons_;
   ButtonSubscriptionsMap pending_button_subscriptions_;
+  ButtonSubscriptionsMap pending_button_unsubscriptions_;
```

Comment:
@VadymLuchko maybe it's possible to replace these 2 maps with one which will work with the buttons states?
For example declare:
```
enum class ButtonSubscriptionState {
  kNotSubscribed,
  kPendingSubscription,
  kSubscribed,
  kPendingUnsubscription
}
```
And then just use the tuple< correlation_id, button_name, ButtonSubscriptionState > ?

Reply:
@AKalinich-Luxoft 
formalized it as a separate task
https://adc.luxoft.com/jira/browse/FORDTCN-12269